libs/pageserver_api/src/models.rs

```diff
@@ -227,6 +230,39 @@ pub struct TimelineInfo {
     pub state: TimelineState,
 }
 
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Serialize, Deserialize)]
+pub struct LayerMapInfo {
+    pub in_memory_layers: BTreeSet<InMemoryLayerInfo>,
+    pub historic_layers: BTreeSet<HistoricLayerInfo>,
+}
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Copy, Clone, Serialize, Deserialize)]
+#[serde(tag = "kind")]
+pub enum InMemoryLayerInfo {
+    Open { lsn_start: Lsn },
+    Frozen { lsn_start: Lsn, lsn_end: Lsn },
+}
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Serialize, Deserialize)]
```

Comment:
The `Ord` is annoying here because we want to include the layer access stats in this struct.
I suppose we need it for sorted output. Can we just not sort the output and let the client deal with this?

Reply:
Similarly Eq and PartialEq

Reply:
Will prepare a patch

Reply:
https://github.com/neondatabase/neon/pull/3486#discussion_r1091712440